engines/router/missionctl/handlers/logutils.go

```diff
@@ -44,16 +45,21 @@ func logTuringRouterRequestSummary(
 	for resp := range mcRespCh {
 		// If error exists, add an error record
 		if resp.err != "" {
-			logEntry.AddResponse(resp.key, nil, resp.err)
+			logEntry.AddResponse(resp.key, nil, nil, resp.err)
 		} else {
 			// Process the response body
 			uncompressedData, err := uncompressHTTPBody(resp.header, resp.body)
 			if err != nil {
 				logger.Errorf("Error occurred when reading %s response body: %s",
 					resp.key, err.Error())
-				logEntry.AddResponse(resp.key, nil, err.Error())
+				logEntry.AddResponse(resp.key, nil, nil, err.Error())
 			} else {
-				logEntry.AddResponse(resp.key, uncompressedData, "")
+				// Format the response header
+				responseHeader := map[string]string{}
```

Comment:
Minor: The `NewTuringResultLogEntry` function in the `resultlog` package has the same logic for formatting headers. Can we extract that into a separate function that can also be reused here?

Reply:
Okay that's done :)